stream-chat-swift-docs/docs/guides/working-with-user.md

```diff
@@ -1,3 +1,136 @@
 ---
 title: Working with User
----
\ No newline at end of file
+---
+
+The User object is accessed through a `Controller`. 
+There is a special type of user object which is `CurrentUser`, representing the currently signed-in user. 
+
+```swift
+/// User id of the user you want to work with
+let userId = "yourUserId"
+
+/// User controller for the intended user
+let userController = chatClient.usersController(userId: userId)
+
+/// Current user controller
+let currentUserController = chatClient.currentUserController()
+```
+
+## Standard User
+
+`CharUser` is a simple object, you can access the information and (un)mute and (un)flag the user.
+```swift
+let user = userController.user
+print(user.name)
+```
+
+### Muting/unmuting user
+```swift 
+userController.mute()
+userController.unmute()
+```
+Both functions have an optional completion block to be called when the network request is finished.
+```swift 
+userController.mute {
+    if let error = error {
+        print(error)
+        return
+    }
+    // Successfully finished
+}
+```
+
+### Flagging/Unflagging user
+```swift 
+userController.flag()
+userController.unflag()
+```
+Both functions have an optional completion block to be called when the network request is finished.
+```swift 
+userController.flag {
+    if let error = error {
+        print(error)
+        return
+    }
+    // Successfully finished
+}
+```
+
+## Updating current user data
+
+You can call update user data function on the current user controller.
+
+```swift 
+currentUserController.updateUserData(name: "Luke Skywalker")
+```
+
+You can observe the changes in user data with delegate.
+
+```swift 
+// Don't forget to keep reference to the delegate. 
+// `CurrentUserController` keeps only weak reference to the delegate.
```

Comment:
Let's remove these

Reply:
I know I struggled with this when I was a newbie dev (my first year), I spend more than I would like on debugging issues like this 😅 
What is the benefit of removing those?

Reply:
We don't want to let guides dive too deep into development basics and divert from the actual point. Our delegates work like any other delegate, we should only document what we're doing different